tiles/src/main/java/com/protomaps/basemap/postprocess/Clip.java

```diff
@@ -0,0 +1,155 @@
+package com.protomaps.basemap.postprocess;
+
+import static com.onthegomap.planetiler.geo.GeoUtils.WORLD_BOUNDS;
+import static com.onthegomap.planetiler.geo.GeoUtils.latLonToWorldCoords;
+import static com.onthegomap.planetiler.render.TiledGeometry.getCoveredTiles;
+import static com.onthegomap.planetiler.render.TiledGeometry.sliceIntoTiles;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.onthegomap.planetiler.ForwardingProfile;
+import com.onthegomap.planetiler.VectorTile;
+import com.onthegomap.planetiler.geo.*;
+import com.onthegomap.planetiler.render.TiledGeometry;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+import com.onthegomap.planetiler.stats.Stats;
+import com.protomaps.basemap.GeoJSON;
+import org.locationtech.jts.geom.*;
+import org.locationtech.jts.geom.util.AffineTransformation;
+import org.locationtech.jts.operation.overlayng.OverlayNG;
+import org.locationtech.jts.operation.overlayng.OverlayNGRobust;
+
+public class Clip implements ForwardingProfile.TilePostProcessor {
+  private final Map<Integer, Map<TileCoord, List<List<CoordinateSequence>>>> tiledGeometries;
+  private final Map<Integer, TiledGeometry.CoveredTiles> coverings;
+  private final Stats stats;
+
+  public Clip(Stats stats, Geometry input) {
+    this.stats = stats;
+    var clipGeometry = latLonToWorldCoords(input).buffer(0.00001);
+    tiledGeometries = new HashMap<>();
+    coverings = new HashMap<>();
+    try {
+      for (var i = 0; i <= 15; i++) {
+        var extents = TileExtents.computeFromWorldBounds(i, WORLD_BOUNDS);
+        double scale = 1 << i;
+        Geometry scaled = AffineTransformation.scaleInstance(scale, scale).transform(clipGeometry);
+//        var simplified = DouglasPeuckerSimplifier.simplify(scaled, 0.25/256);
```

Comment:
Do we need to simplify, or rely on the last snapping step?

Reply:
I would be surprised if we had to.